archetype/engine-v2/src/main/java/io/helidon/build/archetype/engine/v2/ArchetypeEngineV2.java

```diff
@@ -53,6 +58,7 @@ private ArchetypeEngineV2(Builder builder) {
         this.onResolved = builder.onResolved;
         this.directorySupplier = builder.directorySupplier;
         this.outputPropsFile = builder.outputPropsFile;
+        this.cliVersion = builder.cliVersion;
```

Comment:
The versions enforced here are schema versions.
The engine needs to define a constant for the schema version it supports. The archetypes define a schema version attribute (MANIFEST.MF) ; since an archetype is a collection of many XML files, the manifest attribute must be the greatest version found in all the files.

Reply:
I see, so `helidon-archetype-maven-plugin` will look for the maximum schema version among all xml files and write it to  MANIFEST.MF during package phase, right? 
The questions are as follows: how the schema version will be stored in the archetype xml files and how  this schema version will be related to the versions in `versions.xml`?

Reply:
The schema version is declared in the schema URL:

```xml
<archetype-script xmlns="https://helidon.io/archetype/2.0"
                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                  xsi:schemaLocation="https://helidon.io/archetype/2.0 file:/archetype.xsd">
```

In the snippet above it would be `2.0`.

---

The schema version is not related to `versions.xml`. The only relationship is that a cli includes an engine that supports a certain schema version ; however that information is not recorded in `versions.xml`.